pkg/agent/globalstate.go

```diff
@@ -507,6 +517,23 @@ func (s *lockedPodStatus) update(global *agentState, with func(podStatus) podSta
 	s.podStatus = newStatus
 }
 
+func (s podStatus) isStuck(global *agentState, now time.Time) bool {
+	if s.monitorStuckAt(global.config).Before(now) {
+		return true
+	}
+	if s.failedMonitorRequestCounter.Get() > global.config.Monitor.MaxFailedRequestRate.Threshold {
+		return true
+	}
+	if s.failedSchedulerRequestCounter.Get() > global.config.Scheduler.MaxFailedRequestRate.Threshold {
+		return true
+	}
+	if s.failedNeonVMRequestCounter.Get() > global.config.NeonVM.MaxFailedRequestRate.Threshold {
+		return true
+	}
+	return false
```

Comment:
IMO this should just be a single large statement, something like
```go
return s.monitorStuckAt(global.config).Before(now) ||
    s.failedMonitorRequestCounter.Get() > global.config.Monitor.MaxFailedRequestRate.Threshold ||
    ...
```


wdyt? my thinking is that visually, it should be easier to parse, and it's syntactically more clear

Reply:
Alternatively, it'd be good to include information in the logs (and maybe metrics?) about _why_ the VM is stuck - changing `isStuck` to return `[]string` would make a lot of sense, I think.

Reply:
> to return []string

Or a binary mask.

Reply:
Added reason here 20a66f353704170fae0f1aa760905f4ce3674a1e.